src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,123 @@
+package duke;
+
+import duke.task.Task;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class TaskList {
+    final static BadCommandException INDEX_OOB_ERROR =
+            new BadCommandException("Index given is out of bounds!");
+    protected ArrayList<Task> tasks;
```

Comment:
I feel like this should be private in accordance to Java guidelines. There are multiple of these instances later on as well. Since the methods to modify the class instance variables should be done within class methods itself.